prov/util/src/util_attr.c

```diff
@@ -47,8 +47,8 @@ int ofi_valid_addr_format(uint32_t prov_format, uint32_t user_format)
 
 	switch (prov_format) {
 	case FI_SOCKADDR:
-		/* Provider supports INET and INET6 */
-		return user_format <= FI_SOCKADDR_IN6;
+		/* Provider supports IB, INET, and INET6 */
+		return user_format <= FI_SOCKADDR_IB;
```

Comment:
This change makes FI_SOCKADDR equivalent to FI_SOCKADDR_IB.  I wonder if the provider shouldn't be using FI_SOCKADDR_IB instead of FI_SOCKADDR.  My concern here is that providers that support IPv4 and IPv6 only (udp, tcp, net) will now end up accepting IB addresses.

Reply:
You mean change this line https://github.com/ofiwg/libfabric/blob/main/prov/verbs/src/verbs_info.c#L774
and have `fi->addr_format = FI_SOCKADDR_IB;`?

Reply:
Yes, that is what I meant.  It looks like that would align the code with the comments in this function.  But let me look through the rest of the code to see what makes sense.  Your change makes sense from reading the API; I'm just unsure if that's what we want.  (FI_SOCKADDR appears to be used more like FI_SOCKADDR_INX or FI_SOCKADDR_IP.)

Reply:
Understood, thanks! 
I will run some tests with your suggestion and post my results here.

Reply:
I looked through the code, and if changing verbs_info.c like you mentioned above works, that's the route we should take.  There's no clean way for a provider to indicate that they only support IP addressing (needed by udp, net, tcp), so they use FI_SOCKADDR in their fi_getinfo filter.

Technically, those providers really do support a generic sockaddr structure, limited by the kernel.  But if we can filter at fi_getinfo, then that prevents an app from trying to use, say, sockaddr_ib with the tcp provider and dealing with the failure at endpoint creation.

A better longer term solution would be to define, say, FI_SOCKADDR_IP or OFI_SOCKADDR_IP and using that as a more selective filter.  The 'OFI' definition would be if we want to keep this internal.

Reply:
I tested the change `fi->addr_format = FI_SOCKADDR_IB` in the verbs provider and it works for my use case! 
I have updated the patch accordingly. Thank you!